components/hab/src/main.rs

```diff
@@ -1563,60 +1564,73 @@ fn print_svc_status<T>(out: &mut T,
                        -> result::Result<(), SrvClientError>
     where T: io::Write
 {
-    let status = match reply.message_id() {
+    match reply.message_id() {
         "ServiceStatus" => {
-            reply.parse::<sup_proto::types::ServiceStatus>()
-                 .map_err(SrvClientError::Decode)?
+            let status = reply.parse::<sup_proto::types::ServiceStatus>()
+                              .map_err(SrvClientError::Decode)?;
+            let svc_desired_state = status.desired_state
```

Comment:
Since this is a big chunk of code that's mostly moved, but also has some other changes, I find it makes it easier to review if broken into separate commits:
1. Move existing code to new location with no logic changes
1. Apply logic changes